docs/examples/spectrum/spectral_factor.py

```diff
@@ -0,0 +1,138 @@
+#   :math:`AM_a` and the atmospheric precipitable water content :math:`W`
+
+# %%
+# Calculation of inputs
+# ---------------------
+# Let's calculate the absolute air mass, which is required for all three
+# models. We use the Kasten and Young [2]_ model, which requires the apparent
+# sun zenith. Note: TMY3 files timestamps indicate the end of the hour, so we
+# shift the indices back 30-minutes to calculate solar position at the centre
+# of the interval.
+
+# Create a location object
+lat, lon = metadata['latitude'], metadata['longitude']
+alt = altitude = metadata['altitude']
+tz = 'Etc/GMT+5'
+loc = location.Location(lat, lon, tz=tz, name='Greensboro, NC')
+
+# Calculate solar position parameters
+solpos = loc.get_solarposition(
+    meteo.index.shift(freq="-30min"),
+    pressure=meteo.pressure*100,  # convert from millibar to Pa
+    temperature=meteo.temp_air)
+solpos.index = meteo.index  # reset index to end of the hour
+
+amr = get_relative_airmass(solpos.apparent_zenith).dropna()  # relative airmass
+ama = amr*(meteo.pressure/1013.25)  # absolute airmass
+
+# Now we calculate the clearsky index, :math:`kc`, by comparing the TMY3 GHI to
+# the modelled clearky GHI.
+
+cs = loc.get_clearsky(meteo.index)
+kc = pv.irradiance.clearsky_index(meteo.ghi, cs.ghi)
+
+# :math:`W` is provided in the TMY3 file but in other cases can be calculated
+# from temperature and relative humidity
+# (see :py:func:`pvlib.atmosphere.gueymard94_pw`).
+
+w = meteo.precipitable_water
+
+# %%
+# Calculation of Spectral Mismatch
+# --------------------------------
+# Let's calculate the spectral mismatch factor using the three pvlib functions.
+# First, we need to import some model coefficients for the SAPM spectral factor
+# function, which, unlike the other two functions, lacks built-in coefficients.
+
+# Import some for a mc-Si module from the SAPM module database.
+module = pv.pvsystem.retrieve_sam('SandiaMod')['LG_LG290N1C_G3__2013_']
+# Calculate M using the three models for an mc-Si PV module.
+m_sapm = pv.spectrum.spectral_factor_sapm(ama, module)
+m_pvspec = pv.spectrum.spectral_factor_pvspec(ama, kc, 'multisi')
+m_fs = pv.spectrum.spectral_factor_firstsolar(w, ama, 'multisi')
+df_results = pd.concat([m_sapm, m_pvspec, m_fs], axis=1)
+df_results.columns = ['SAPM', 'PVSPEC', 'FS']
+# %%
+# Comparison Plots
+# ----------------
+# We can plot the results to visualise variability between the models. Note
+# that this is not an exact comparison since the exact same PV modules has
+# not been modelled in each case, only the same module technology.
+
+# Plot M
+fig1, ax1 = plt.subplots()
+df_results.plot(ax=ax1)
+
+ax1.set_xlabel('Date (m-d H:M)')
+ax1.set_ylabel('Mismatch (-)')
+ax1.legend()
+ax1.set_ylim(0.85, 1.15)
+plt.show()
+
+# We can also zoom in one one day, for example August 1st.
+fig2, ax1 = plt.subplots()
+df_results.loc['2001-08-01'].plot(ax=ax1)
+ax1.set_ylim(0.85, 1.15)
+plt.show()
```

Comment:
I did not like the original axis labels at first but then I started to prefer that more minimalist style, at least for this simple example... looks less cluttered. I accept the figures might appear too small for some users though. I have tried increasing the size with `plt.figsize()` so let's see how that turns out once the docs load.

Reply:
I tried a range of different ways of plotting...
In the end, although I did not follow your suggestion exactly I did use a few of the formatting techniques that you had adopted (thanks!) and the graphs are now bigger.
Could you please take a look at the graphs as of the latest docs build [0af9b2d](https://github.com/pvlib/pvlib-python/pull/2114/commits/0af9b2d9e81b998aa8ddb73c14afe78b20a750de) and let me know what you think? @IoannisSifnaios

Reply:
It looks much better @RDaxini, nice work! One last thing about this, since both figures have the same legend, you can put the legend outside  of the figure using `bbox_to_anchor`.

So you could try something like:
`ax1.legend(loc='upper center', frameon=False, ncols=3, bbox_to_anchor=(0.5, 1.3))`

and not have any legend for ax2

Reply:
> Since both figures have the same legend, you can put the legend outside of the figure using `bbox_to_anchor`.
> 
> So you could try something like: `ax1.legend(loc='upper center', frameon=False, ncols=3, bbox_to_anchor=(0.5, 1.3))` and not have any legend for ax2

Nice one

> And maybe remove the title from the bottom subplot (?). I think it is clear from the axis labels that it is for one day...

You mean "day"? That is not a title of the bottom subplot, it is  the axis label of the upper subplot 😅
It looks a bit low because it is written below the line of "2001"